ZeeHomeScreen/Classes/Components/SectionComposite/SectionCompositeViewController.swift

```diff
@@ -274,27 +274,43 @@ import Zee5CoreSDK
         }
         toggle.setButtonsTitles(buttonsTitles: ["nav_livetv".localized(hashMap: [:]), "EPG_Header_ChannelGuide_Text".localized(hashMap: [:])])
         toggle.changeToIndex = { [weak self] index in
-            self?.collectionView?.isHidden = index == 0 ? false : true
-            self?.epgContentView?.isHidden = index == 0 ? true : false
+            switch index {
+            case 0:
+                self?.collectionView?.isHidden = false
+                self?.epgContentView?.isHidden = true
+
+            case 1:
+                self?.prepareEPGViewIfNeeded()
```

Comment:
Don't prepare EPG view while loading the screen, but rather when used clicks on EPG button